app/decorators/auth_data_decorator.rb

```diff
@@ -0,0 +1,4 @@
+class AuthDataDecorator < ApplicationDecorator
+  delegate :provider, :uid, :info
```

Comment:
Are you sure that delegate will work? As far as I understand `request.env["omniauth.auth"]`
 will return hash
```
{	
      email: authentication_hash.dig(:info, :email),	
      full_name: authentication_hash.dig(:info, :name),	
      uid: authentication_hash.dig(:uid),	
      provider: authentication_hash.dig(:provider)	
    }
```

Reply:
It is a object of class ```OmniAuth::AuthHash```, it is allow to get hash values by calling method, where method name is a key name.